src/components/CivitaiWrapped/CivitaiSessionProvider.tsx

```diff
@@ -1,50 +1,71 @@
 import { onboardingSteps } from '~/components/Onboarding/onboarding.utils';
 import { useDomainSync } from '~/hooks/useDomainSync';
 import { useFeatureFlags } from '~/providers/FeatureFlagsProvider';
 import { UserMeta } from '~/server/schema/user.schema';
-import { nsfwBrowsingLevelsFlag } from '~/shared/constants/browsingLevel.constants';
+import {
+  browsingModeDefaults,
+  nsfwBrowsingLevelsFlag,
+} from '~/shared/constants/browsingLevel.constants';
 import { Flags } from '~/shared/utils';
-
+import { useCookies } from '~/providers/CookiesProvider';
+import { deleteCookie } from 'cookies-next';
 const OnboardingModal = dynamic(() => import('~/components/Onboarding/OnboardingWizard'));
 
 export function CivitaiSessionProvider({ children }: { children: React.ReactNode }) {
   const { data, update } = useSession();
-  const { useStore } = useBrowsingModeContext();
-  const browsingModeState = useStore((state) => state);
+  const user = data?.user;
   const { canViewNsfw } = useFeatureFlags();
+  const cookies = useCookies();
   useDomainSync(data?.user as SessionUser);
 
-  const value = useMemo(() => {
-    if (!data?.user) return null;
-
-    return {
-      ...data.user,
-      isMember: data.user.tier != null,
-      memberInBadState: data.user.memberInBadState,
+  const { disableHidden } = cookies;
+
+  const sessionUser = useMemo(() => {
+    if (!user)
+      return {
+        type: 'unauthed',
+        settings: publicContentSettings,
+      } as UnauthedUser;
+
+    const isMember = user.tier != null;
+    const currentUser: AuthedUser = {
+      type: 'authed',
+      ...user,
+      isMember,
+      memberInBadState: user.memberInBadState,
       refresh: update,
-      ...(!canViewNsfw
-        ? { showNsfw: false, blurNsfw: true, browsingLevel: 1 }
-        : {
-            ...browsingModeState,
-            blurNsfw: !Flags.intersection(browsingModeState.browsingLevel, nsfwBrowsingLevelsFlag)
-              ? true
-              : browsingModeState.blurNsfw,
-          }),
+      settings: {
+        showNsfw: user.showNsfw,
+        browsingLevel: user.browsingLevel,
+        disableHidden: disableHidden ?? true,
+        allowAds: user.allowAds ?? !isMember ? true : false,
+        autoplayGifs: user.autoplayGifs ?? true,
+        blurNsfw: !Flags.intersection(user.browsingLevel, nsfwBrowsingLevelsFlag)
+          ? true
+          : user.blurNsfw,
+      },
     };
-  }, [data?.expires, update, browsingModeState]); // eslint-disable-line
+    if (!canViewNsfw) currentUser.settings = { ...currentUser.settings, ...browsingModeDefaults };
+    return currentUser;
+  }, [data?.expires, disableHidden, canViewNsfw]);
 
   useEffect(() => {
     if (data?.error === 'RefreshAccessTokenError') signIn();
   }, [data?.error]);
 
   useEffect(() => {
-    const onboarding = value?.onboarding;
+    deleteCookie('level');
+    deleteCookie('blur');
+    deleteCookie('nsfw');
+  }, []);
```

Comment:
Do we want to remove these on every refresh?

Reply:
Does it hurt? I just want to ensure that these cookies get removed since we are no longer using them.